src/VisualStudio/IntegrationTest/TestUtilities/InProcess/ErrorList_InProc.cs

```diff
@@ -30,15 +30,20 @@ public void WaitForNoErrorsInErrorList()
             }
         }
 
-        public void NavigateToErrorListItem(int itemIndex)
+        public ErrorListItem NavigateToErrorListItem(int itemIndex, __VSERRORCATEGORY minimumSeverity = __VSERRORCATEGORY.EC_WARNING)
```

Comment:
I kinda wish this wasn't a default. Otherwise it's *really* surprising that NavigateToErrorListItem(0) isn't actually "navigate to the first item".

Reply:
Also, any reason this isn't just a precise filter? If I'm trying to navigate to a warning, it'll still randomly choose an error as well? I should know the type I'm trying to go to, why not just filter to exact severity?

Reply:
The filtering was added as part of switching from DTE to the extensibility interfaces. The DTE operations do not return results that are filtered out, and the default error list view only has Errors and Warnings checked. However, this was problematic for local developers where the experimental instance might have a different set of items checked (say just Errors).

I'm not sure what the best approach is, but the current filtering defaults of warnings seem to match what the code was expecting with slightly less flakiness than was originally present.

Reply:
It seems like the most reliable way would be to navigate to the first item with a specific error code. That way it wouldn't be impacted by either other changes or the order in which different error providers populate the list.

Reply:
@sharwell That seems wisest. Can we try that?

Reply:
Here's the skeleton of my last try 😄 

https://github.com/dotnet/roslyn/blob/4f74a300efc91b8010b045fcb8cfc215688418f2/src/VisualStudio/IntegrationTest/TestUtilities/InProcess2/ErrorList_InProc2.cs#L136-L147